src/features/courses/course-controller.ts

```diff
@@ -2224,6 +2251,49 @@ class CourseController {
         });
     }
 
+    async getGradeForQuestion(options: GetGradeForQuestionOptions): Promise<StudentGrade> {
+        const {
+            questionId,
+            userId,
+            includeWorkbooks
+        } = options;
+        const include: sequelize.IncludeOptions[] = [{
+            model: StudentGradeInstance,
+            as: 'gradeInstances',
+            required: false,
+            where: {
+                active: true,
+            }
+        }];
+        if (includeWorkbooks) {
+            include.push({
+                model: StudentWorkbook,
+                as: 'workbooks',
+                required: false,
+                where: {
+                    active: true
+                }
+            });
+        }
+        try {
+            const grade = await StudentGrade.findOne({
+                where: {
+                    courseWWTopicQuestionId: questionId,
+                    userId,
+                    active: true,
+                },
+                include
+            });
+            if (!_.isNil(grade)) {
+                return grade;
+            } else {
+                throw new Error('Did not find a student grade.');
```

Comment:
This should be rederly error, it helps in debugging and handling to know that the error came from within the application

Furthermore shouldn't this be an illegal argument except because this can be exercised by the API and is not an unhandled exception